src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -319,6 +320,63 @@ mgp_value_type FromTypedValueType(memgraph::query::TypedValue::Type type) {
 }
 }  // namespace
 
+bool MgpGraphIsTransactional(const mgp_graph &graph) noexcept {
+  auto storageMode = memgraph::storage::StorageMode::IN_MEMORY_TRANSACTIONAL;
+  std::visit(memgraph::utils::Overloaded{
+                 [&storageMode](memgraph::query::DbAccessor *dbAcc) { storageMode = dbAcc->GetStorageMode(); },
+                 [&storageMode](memgraph::query::SubgraphDbAccessor *dbAcc) { storageMode = dbAcc->GetStorageMode(); }},
+             graph.impl);
+  return storageMode != memgraph::storage::StorageMode::IN_MEMORY_ANALYTICAL;
+}
```

Comment:
Due to complex visit logic and complex logic here overall, I am more prone to solution where we move this check inside Storage. Also, one thing that popped on my mind now is that we can cache this info in mgp_graph during mgp_graph object creation as this will not change during procedure work, and we have benefit of not checking for every object if it was deleted or not every time

Reply:
Done